test/metabase/api/collection_test.clj

```diff
@@ -639,6 +640,17 @@
                 (:data (mt/user-http-request :crowberto :get 200
                                              (str "collection/" (u/the-id collection) "/items"))))))))))
 
+(deftest collection-items-based-on-upload-test
+  (testing "GET /api/collection/:id/items"
+    (testing "check that based_on_upload is returned for cards correctly"
+      (api.card-test/run-based-on-upload-test
+       (fn [card]
+         (->> (mt/user-http-request :crowberto :get 200 (str "collection/" (:collection_id card) "/items"))
```

Comment:
you either should add `models=dataset` to the query param`, or you need to filter by model below.

Reply:
why do you suggest this? I don't want to do this because we need to test responses for cards as well as datasets

Reply:
to make sure the thing you filtered by id is a model, not sth that accidentally have the same `id`

Reply:
true, updated in [8e21f21](https://github.com/metabase/metabase/pull/38322/commits/8e21f2140ad8a6b87258456d7001bf309aa1b0e7)